### PR TITLE
add professional/creators/profiles/analytics API

### PR DESCRIPTION
**Pull request by vasu-iiq**
## Old screenshots
<details>
image
image
image
</details>

## New screenshots
<details>
image
image
image
</details>

### CODE REVIEW

**Review by anujith-s — DISMISSED** [review]

File: app/events/db/db_table_field_mappings.yaml [hunk 1]
```diff
@@ -608,3 +608,43 @@ mappings:
             profile_following_count: app.events.db.value_post_processors.cast_string_to_int
             profile_subscriber_count: app.events.db.value_post_processors.cast_string_to_int
             contact_details: app.events.db.value_post_processors.cast_to_json
+
+
+    profile_analytics:
```
Comment (vasu-iiq) [inline app/events/db/db_table_field_mappings.yaml:613]: Check: Getting an error in Pycharm here on duplicate name. Working fine via Postman.

File: app/events/db/migrations/db_table_sql_scripts.sql [hunk 2]
```diff
@@ -622,3 +622,36 @@ CREATE TABLE IF NOT EXISTS iiq_schema.profile_contact_info
 );
 ALTER TABLE iiq_schema.profile_contact_info
     OWNER TO iiq;
+
+
+CREATE TABLE IF NOT EXISTS iiq_schema.profile_analytics
```
Comment (ghost) [inline app/events/db/migrations/db_table_sql_scripts.sql]: I think, you should rename it to professional_profile_analytics

File: app/events/db/db_table_field_mappings.yaml (hunk 1 above)
Comment (ghost) [inline app/events/db/db_table_field_mappings.yaml]: I think, you should rename it to professional_profile_analytics

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by anujith-s — DISMISSED** [review]

File: app/events/db/migrations/db_table_sql_scripts.sql [hunk 3]
```diff
@@ -622,3 +622,36 @@ CREATE TABLE IF NOT EXISTS iiq_schema.profile_contact_info
...
+    location_country       VARCHAR(255),
+    languages              JSONB,
+    work_experiences       JSONB,
+    education              JSONB,
+    publications           JSONB,
+    certifications         JSONB,
+    volunteer_experiences  JSONB,
+    honors                 JSONB,
+    projects               JSONB,
+    CONSTRAINT unique_profile UNIQUE (work_platform_id, platform_username)
```
Comment (ghost) [inline app/events/db/migrations/db_table_sql_scripts.sql]: change the name of constraint

**Review by ghost — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

File: app/services/resource_service.py [hunk 4]
```diff
@@ -423,3 +423,18 @@ async def fetch_contact_info(request_body: object, params: Optional[Dict]) -> Di
     # TODO do error-handling over here
 
     return response
+
+
+@retry(attempts=3, delay=1, retry_exceptions=(TooManyRequestException,))
```
Comment (shahaab-iiq) [inline app/services/resource_service.py:428]: why extra comma in the end?
@retry(attempts=3, delay=1, retry_exceptions=(TooManyRequestException,))

**Review by shahaab-iiq — APPROVED** [review]

File: app/services/resource_service.py (hunk 4 above)
Comment (vasu-iiq) [reply app/services/resource_service.py:428]: That's actually the convention.
This syntax is specifically used to define a tuple with a single element.

**Pull request merged**